app/models/queue_item.rb

```diff
@@ -0,0 +1,151 @@
+class QueueItem < ActiveRecord::Base
+  belongs_to :user
+  belongs_to :video
+  before_validation :set_position, unless: :has_position?
+
+  delegate :category, to: :video
+
+  def rating
+    review.try(:rating)
+  end
+
+  def destroy_by_user(the_user)
+    if user == the_user
+      destroy
+      QueueItem.normalize(user)
+      true
+    else
+      false
+    end
+  end
+
+  def update_rating(rating)
+    if review
+      review.update_attributes(rating: rating)
```

Comment:
You could use `.update` instead of `update_attributes` its shorter and does the same thing.